### DISCUSSION TITLE
Linux Mint Mate - Unable to run Streamdeck-UI - core dumped

### DISCUSSION THREAD

**Discussion by fliptoback** [streamdeck-linux-gui/streamdeck-linux-gui#129] [Q&A] [answered]
Hi everyone,

I am trying to use streamdeck-UI on my linux mint mate. I have installed based on the instructions but it gives me a core dumped.  I am on Linux Mint 21.2 using kernel 6.2.0.37.

The error message is as follows.

`qt.qpa.plugin: From 6.5.0, xcb-cursor0 or libxcb-cursor0 is needed to load the Qt xcb platform plugin.
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vkkhrdisplay, vnc, wayland-egl, wayland, xcb.

Aborted (core dumped)
`
Looking for feedback how I can get this installed please?

Thanks in advance.

**Comment by coolapso** [comment]
That's a wayland / desktio environment related issue, try starting it from the terminal like this and let met know if it works.

`QT_QPA_PLATFORM=wayland streamdeck`


What login manager and desktop environment are you using?

**Reply by fliptoback** [reply]
Hi thanks for replying.

I am not sure if I am running your command correctly - but running QT_QPA_PLATFORM=wayland streamdeck on the commandline gives me this error.

`Failed to create wl_display (No such file or directory)
qt.qpa.plugin: Could not load the Qt platform plugin "wayland" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vkkhrdisplay, vnc, wayland-egl, wayland, xcb.

Aborted (core dumped)
`
I am running Linux Mint Mate 21.2.

AMD Ryzen 5 5600X, 64GB ram and AMD RX6600 GPU
Kernel 6.2.0-37-generic
Mate v1.26.0

Any ideas?

**Reply by coolapso** [reply]
Interesting ...  what is wayland using then .. i've seen this error when that variable is not exported .... what does 

```echo $XDG_SESSION_TYPE```  returns to you? 

do you have `libxcb` installed? 

maybe some `apt search libxcb`  should give you some hints

**Reply by fliptoback** [reply]
I dont think I am running wayland - pretty sure Linux Mint is an X11 system.

Edit: previously when i was using Linux Mint Cinnamon - Streamdeck UI is working - and Linux Mint Cinnamon is also X11 based.

running "echo $XDG_SESSION_TYPE"  return the string "X11".

running apt search libxcb return this:

- `i A libxcb-composite0               - X C Binding, composite extension          
- p   libxcb-composite0:i386          - X C Binding, composite extension          
- p   libxcb-composite0-dev           - X C Binding, composite extension, developm
- p   libxcb-composite0-dev:i386      - X C Binding, composite extension, developm
- p   libxcb-cursor-dev               - utility libraries for X C Binding -- curso
- p   libxcb-cursor0                  - utility libraries for X C Binding -- curso
- i A libxcb-damage0                  - X C Binding, damage extension             
- p   libxcb-damage0:i386             - X C Binding, damage extension             
- p   libxcb-damage0-dev              - X C Binding, damage extension, development
- p   libxcb-damage0-dev:i386         - X C Binding, damage extension, development
- p   libxcb-doc                      - X C Binding, development documentation    
- v   libxcb-doc:i386                 -                                           
- p   libxcb-dpms0                    - X C Binding, dpms extension               
- p   libxcb-dpms0:i386               - X C Binding, dpms extension               
- p   libxcb-dpms0-dev                - X C Binding, dpms extension, development f
- p   libxcb-dpms0-dev:i386           - X C Binding, dpms extension, development f
- i   libxcb-dri2-0                   - X C Binding, dri2 extension               
- p   libxcb-dri2-0:i386              - X C Binding, dri2 extension               
- p   libxcb-dri2-0-dev               - X C Binding, dri2 extension, development f
- p   libxcb-dri2-0-dev:i386          - X C Binding, dri2 extension, development f
- i   libxcb-dri3-0                   - X C Binding, dri3 extension               
- p   libxcb-dri3-0:i386              - X C Binding, dri3 extension               
- p   libxcb-dri3-dev                 - X C Binding, dri3 extension, development f
- p   libxcb-dri3-dev:i386            - X C Binding, dri3 extension, development f
- p   libxcb-ewmh-dev                 - utility libraries for X C Binding -- ewmh,
- p   libxcb-ewmh-dev:i386            - utility libraries for X C Binding -- ewmh,
- p   libxcb-ewmh2                    - utility libraries for X C Binding -- ewmh 
- p   libxcb-ewmh2:i386               - utility libraries for X C Binding -- ewmh 
- i   libxcb-glx0                     - X C Binding, glx extension                
- p   libxcb-glx0:i386                - X C Binding, glx extension                
- p   libxcb-glx0-dev                 - X C Binding, glx extension, development fi
- p   libxcb-glx0-dev:i386            - X C Binding, glx extension, development fi
- i   libxcb-icccm4                   - utility libraries for X C Binding -- icccm
- p   libxcb-icccm4:i386              - utility libraries for X C Binding -- icccm
- p   libxcb-icccm4-dev               - utility libraries for X C Binding -- icccm
- p   libxcb-icccm4-dev:i386          - utility libraries for X C Binding -- icccm
- i   libxcb-image0                   - utility libraries for X C Binding -- image
- p   libxcb-image0:i386              - utility libraries for X C Binding -- image
- p   libxcb-image0-dev               - utility libraries for X C Binding -- image
- p   libxcb-image0-dev:i386          - utility libraries for X C Binding -- image
- p   libxcb-imdkit-dev               - XIM protocol implementation in XCB (develo
- p   libxcb-imdkit1                  - XIM protocol implementation in XCB        
- i   libxcb-keysyms1                 - utility libraries for X C Binding -- keysy
- p   libxcb-keysyms1:i386            - utility libraries for X C Binding -- keysy
- p   libxcb-keysyms1-dev             - utility libraries for X C Binding -- keysy
- p   libxcb-keysyms1-dev:i386        - utility libraries for X C Binding -- keysy
- p   libxcb-present-dev              - X C Binding, present extension, developmen
- p   libxcb-present-dev:i386         - X C Binding, present extension, developmen
- i   libxcb-present0                 - X C Binding, present extension            
- p   libxcb-present0:i386            - X C Binding, present extension            
- i   libxcb-randr0                   - X C Binding, randr extension              
- p   libxcb-randr0:i386              - X C Binding, randr extension              
- p   libxcb-randr0-dev               - X C Binding, randr extension, development 
- p   libxcb-randr0-dev:i386          - X C Binding, randr extension, development 
- p   libxcb-record0                  - X C Binding, record extension             
- p   libxcb-record0:i386             - X C Binding, record extension             
- p   libxcb-record0-dev              - X C Binding, record extension, development
- p   libxcb-record0-dev:i386         - X C Binding, record extension, development
- i   libxcb-render-util0             - utility libraries for X C Binding -- rende
- p   libxcb-render-util0:i386        - utility libraries for X C Binding -- rende
- p   libxcb-render-util0-dev         - utility libraries for X C Binding -- rende
- p   libxcb-render-util0-dev:i386    - utility libraries for X C Binding -- rende
- i   libxcb-render0                  - X C Binding, render extension             
- p   libxcb-render0:i386             - X C Binding, render extension             
- p   libxcb-render0-dev              - X C Binding, render extension, development
- p   libxcb-render0-dev:i386         - X C Binding, render extension, development
- i A libxcb-res0                     - X C Binding, res extension                
- p   libxcb-res0:i386                - X C Binding, res extension                
- p   libxcb-res0-dev                 - X C Binding, res extension, development fi
- p   libxcb-res0-dev:i386            - X C Binding, res extension, development fi
- p   libxcb-screensaver0             - X C Binding, screensaver extension        
- p   libxcb-screensaver0:i386        - X C Binding, screensaver extension        
- p   libxcb-screensaver0-dev         - X C Binding, screensaver extension, develo
- p   libxcb-screensaver0-dev:i386    - X C Binding, screensaver extension, develo
- i   libxcb-shape0                   - X C Binding, shape extension              
- p   libxcb-shape0:i386              - X C Binding, shape extension              
- p   libxcb-shape0-dev               - X C Binding, shape extension, development 
- p   libxcb-shape0-dev:i386          - X C Binding, shape extension, development 
- i   libxcb-shm0                     - X C Binding, shm extension                
- p   libxcb-shm0:i386                - X C Binding, shm extension                
- p   libxcb-shm0-dev                 - X C Binding, shm extension, development fi
- p   libxcb-shm0-dev:i386            - X C Binding, shm extension, development fi
- p   libxcb-sync-dev                 - X C Binding, sync extension, development f
- p   libxcb-sync-dev:i386            - X C Binding, sync extension, development f
- v   libxcb-sync0-dev                -                                           
- v   libxcb-sync0-dev:i386           -                                           
- i   libxcb-sync1                    - X C Binding, sync extension               
- p   libxcb-sync1:i386               - X C Binding, sync extension               
- p   libxcb-util-dev                 - utility libraries for X C Binding -- atom,
- p   libxcb-util-dev:i386            - utility libraries for X C Binding -- atom,
- p   libxcb-util0-dev                - dummy transitional package for utility lib
- p   libxcb-util0-dev:i386           - dummy transitional package for utility lib
- i   libxcb-util1                    - utility libraries for X C Binding -- atom,
- p   libxcb-util1:i386               - utility libraries for X C Binding -- atom,
- p   libxcb-xf86dri0                 - X C Binding, xf86dri extension            
- p   libxcb-xf86dri0:i386            - X C Binding, xf86dri extension            
- p   libxcb-xf86dri0-dev             - X C Binding, xf86dri extension, developmen
- p   libxcb-xf86dri0-dev:i386        - X C Binding, xf86dri extension, developmen
- i   libxcb-xfixes0                  - X C Binding, xfixes extension             
- p   libxcb-xfixes0:i386             - X C Binding, xfixes extension             
- p   libxcb-xfixes0-dev              - X C Binding, xfixes extension, development
- p   libxcb-xfixes0-dev:i386         - X C Binding, xfixes extension, development
- i   libxcb-xinerama0                - X C Binding, xinerama extension           
- p   libxcb-xinerama0:i386           - X C Binding, xinerama extension           
- p   libxcb-xinerama0-dev            - X C Binding, xinerama extension, developme
- p   libxcb-xinerama0-dev:i386       - X C Binding, xinerama extension, developme
- p   libxcb-xinput-dev               - X C Binding, xinput extension, development
- p   libxcb-xinput-dev:i386          - X C Binding, xinput extension, development
- i   libxcb-xinput0                  - X C Binding, xinput extension             
- p   libxcb-xinput0:i386             - X C Binding, xinput extension             
- p   libxcb-xkb-dev                  - X C Binding, XKEYBOARD extension, developm
- p   libxcb-xkb-dev:i386             - X C Binding, XKEYBOARD extension, developm
- i   libxcb-xkb1                     - X C Binding, XKEYBOARD extension          
- p   libxcb-xkb1:i386                - X C Binding, XKEYBOARD extension          
- p   libxcb-xrm-dev                  - utility functions for the X resource manag
- p   libxcb-xrm0                     - utility functions for the X resource manag
- p   libxcb-xtest0                   - X C Binding, xtest extension              
- p   libxcb-xtest0:i386              - X C Binding, xtest extension              
- p   libxcb-xtest0-dev               - X C Binding, xtest extension, development 
- p   libxcb-xtest0-dev:i386          - X C Binding, xtest extension, development 
- i   libxcb-xv0                      - X C Binding, xv extension                 
- p   libxcb-xv0:i386                 - X C Binding, xv extension                 
- p   libxcb-xv0-dev                  - X C Binding, xv extension, development fil
- p   libxcb-xv0-dev:i386             - X C Binding, xv extension, development fil
- p   libxcb-xvmc0                    - X C Binding, xvmc extension               
- p   libxcb-xvmc0:i386               - X C Binding, xvmc extension               
- p   libxcb-xvmc0-dev                - X C Binding, xvmc extension, development f
- p   libxcb-xvmc0-dev:i386           - X C Binding, xvmc extension, development f
- i   libxcb1                         - X C Binding                               
- p   libxcb1:i386                    - X C Binding                               
- p   libxcb1-dev                     - X C Binding, development files            
- p   libxcb1-dev:i386                - X C Binding, development files    `

**Reply by coolapso** [reply, marked as answer]
hmm ... maybe you're missing this package? `libxcb-cursor0`  can you check if its installed or just try installing it? 

other than that I am not sure if I can help much further ... I have stopped using X a couple of years ago maybe ... .

**Comment by dhtseany** [comment]
*> qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even
though it was found.*
The XCB module was found so it's likely installed.

*> This application failed to start because no Qt platform plugin could be
initialized. Reinstalling the application may fix this problem.*
Do you have qt6 installed?

**Reply by fliptoback** [reply]
Hi Bro, if I run apt search qt it returns a long list - so i think i do have QT installed - though not sure if it is QT6.

**Reply by coolapso** [reply]
libxcb could also have installed when you did it below .... but i'd find it weird mint be shipping without it ...

**Comment by fliptoback** [comment]
> hmm ... maybe you're missing this package? `libxcb-cursor0` can you check if its installed or just try installing it?
> 
> other than that I am not sure if I can help much further ... I have stopped using X a couple of years ago maybe ... .

My bro. u are a legend! I did as you suggested:

apt install libxcb-cursos0 - this install a bunch of packages.
then run
streamdeck

and it works!!

I am pretty sure i follow right to the letter on the streamdeck installation step on github to install it - but not sure why it misses out the libxcb-cursor0 package.

Thankss again!

**Comment by fliptoback** [comment]
> …



> libxcb could also have installed when you did it below .... but i'd find it weird mint be shipping without it ...

Yes super weird - i was coming from Linux Mint Cinnamon previously - and now using MATE - very strange that this was running when previously on Cinnamon it was quite smooth sailing.
